### PR TITLE
[Mono] Intrinsify As* on ARM64 with mini JIT on ARM64

### PR DESCRIPTION
**Pull request by fanyang-mono**
Contributes to #80566

Intrinsify `Vector128.As*` on ARM64 with mini JIT.

### CODE REVIEW

File: src/mono/mono/mini/cpu-arm64.mdesc [hunk 1]
```diff
@@ -504,6 +504,7 @@ xcompare_fp: dest:x src1:x src2:x len:4
 negate: dest:x src1:x len:4
 ones_complement: dest:x src1:x len:4
 xbinop_forceint: dest:x src1:x src2:x len:4
+xcast: dest:x len:4
```
Comment (vargaz) [inline src/mono/mono/mini/cpu-arm64.mdesc:508]: This needs an src1:x and a clob:1 so it doesn't need to emit a move.

**Comment by jandupej** [comment]
LGTM otherwise.

File: src/mono/mono/mini/cpu-arm64.mdesc (hunk 1 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/cpu-arm64.mdesc:508]: fixed

File: src/mono/mono/mini/mini-arm64.c [hunk 2]
```diff
@@ -3665,6 +3665,8 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			}
 			break;
 		}
+		case OP_XCAST:
```
Comment (tannergooding) [inline src/mono/mono/mini/mini-arm64.c:3668]: It's worth noting that `As` isn't really a regular cast. It's a reinterpretation of bits and in RyuJIT there is no node created for it. It's purely an API to satisfy the higher level type systems and for the JIT we completely elide it in importation to guarantee it is zero cost.

**Comment by fanyang-mono** [comment]
CI failures are not related to this PR.

**Review by vargaz — APPROVED** [review]

**Comment by fanyang-mono** [comment]
CI failures are not related to this PR.

**Pull request merged**